### PR TITLE
`.DEREK.yml` V1.0 support - fixes issue #21

### PR DESCRIPTION
**Pull request by rgee0**
Added support for version 1.0 of `.DEREK.yml`, added associated tests and updated README & GET.  Addresses issue #21

Signed-off-by: rgee0 <richard@technologee.co.uk>

Moved the `isAllowed` checks from the individual switch statements in `commentHandler.go` and instead check in `main.go` that the user is a maintainer and the feature is enabled before passing the JSON through to `commentHandler.go`.

Extracted the functionality from `getMaintainers` and `isMaintainer` into `permissionsHandler.go` and refactored to enable testing.

Added a new type `DerekConfig` to unmarshall `.DEREK.yml` into.

Tested via included tests and be deploying onto `rgee0\Test`.

### CODE REVIEW

**Comment by rgee0** [comment]
I'll get CI working first....

Also just spotted some avoidable duplication.  Will ping when sorted.

**Comment by rgee0** [comment]
@alexellis earlier issues ironed out

File: permissionsHandler.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	"strings"
+	"time"
+
+	yaml "gopkg.in/yaml.v2"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/alexellis/derek/types"
+)
+
+const maintainersFileEnv = "maintainers_file"
```
Comment (alexellis) [inline permissionsHandler.go]: I think we could omit this now.

File: permissionsHandler.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+const defaultMaintFile = ".DEREK.yml"
+
+func getEnv(envVar string, assumed string) string {
```
Comment (alexellis) [inline permissionsHandler.go]: can assumed be `default`? Like os.getenv in Python?

**Comment by rgee0** [comment]
Default is a keyword, no?

File: permissionsHandler.go (hunk 2 above)
Comment (rgee0) [reply permissionsHandler.go]: moot point given we're removing `maintainersFileEnv`

Tried `default` prior to assumed but it was linting as a reserved word.

**Comment by rgee0** [comment]
@alexellis updated to remove the maintainers file env var option.

**Comment by alexellis** [comment]
I think we are ready to graduate the handlers into their own folder and package.

**Pull request merged**